### PR TITLE
Add tests for new programming model versions and .NET 8

### PR DESCRIPTION
**Pull request by hossam-nasr**
Adds new unit tests and long-running tests for Python programming model V2, node (JS and TS) programming model V4, and .NET 8

### CODE REVIEW

File: test/nightly/createProjectAndDeploy.test.ts [hunk 1]
```diff
@@ -23,15 +23,18 @@ interface CreateProjectAndDeployTestCase extends ICreateProjectAndDeployOptions
 }
 
 const testCases: CreateProjectAndDeployTestCase[] = [
-    { title: 'JavaScript', ...getJavaScriptValidateOptions(true), createProjectInputs: [/Model V3/], deployInputs: [getRotatingNodeVersion()] },
-    { title: 'TypeScript', ...getTypeScriptValidateOptions(), createProjectInputs: [/Model V3/], deployInputs: [getRotatingNodeVersion()] },
+    { title: 'JavaScript (Model V3)', ...getJavaScriptValidateOptions(true), createProjectInputs: [/Model V3/], deployInputs: [getRotatingNodeVersion()], languageModelVersion: NodeModelVersion.v3 },
```
Comment (nturinski) [inline test/nightly/createProjectAndDeploy.test.ts:26]: Similar to what Alex mentioned before, we should have a constant for Model V3 and Model V4 inputs.

Comment (hossam-nasr) [reply test/nightly/createProjectAndDeploy.test.ts]: Done :)

File: test/api.test.ts [hunk 2]
```diff
@@ -28,14 +28,14 @@ suite(`AzureFunctionsExtensionApi`, () => {
...
-        await runWithInputs('api.createFunction', [language, /Model V3/, functionName], registerOnActionStartHandler, async () => {
+        await runWithInputs('api.createFunction', [language, NodeModelInput[NodeModelVersion.v3], functionName], registerOnActionStartHandler, async () => {
             await api.createFunction({
                 folderPath,
                 suppressOpenFolder: true,
                 templateId: 'HttpTrigger',
                 languageFilter: /Python|C\#|^(Java|Type)Script$/i,
                 functionSettings: { authLevel: 'anonymous' },
-                targetFramework: ['netcoreapp3.1', 'net6.0'] // Will only work on functions api v1.4.0, but won't hurt on v1.3.0
+                targetFramework: ['netcoreapp3.1', 'net6.0', 'net8.0'] // Will only work on functions api v1.4.0, but won't hurt on v1.3.0
```
Comment (nturinski) [inline test/api.test.ts:38]: I think to test what we really have, it should be ['net6.0', 'net7.0', 'net8.0']

File: test/project/createNewProject.test.ts [hunk 3]
```diff
@@ -8,42 +8,42 @@ import { FuncVersion, JavaBuildTool, ProjectLanguage, TemplateSource } from '../
...
+    { ...getPythonValidateOptions('.venv', FuncVersion.v4), inputs: [PythonModelInput[PythonModelVersion.v1], TestInput.UseDefaultValue], languageModelVersion: PythonModelVersion.v1 },
+    { ...getPythonValidateOptions('.venv', FuncVersion.v4, PythonModelVersion.v2), inputs: [PythonModelInput[PythonModelVersion.v2], TestInput.UseDefaultValue], languageModelVersion: PythonModelVersion.v2 },
+    // Custom language tests
+    { ...getCustomValidateOptions(FuncVersion.v4) }
+];
 
 
-/* temporarily disable Java and Ballerina tests until we can install Java on the new pipelines
-    https://github.com/microsoft/vscode-azurefunctions/issues/4210
+/* Temporarily disable Java and Ballerina tests until we can install Ballerina on the new pipelines
```
Comment (nturinski) [inline test/project/createNewProject.test.ts:41]: Why not leave issue link?

**Review by nturinski — DISMISSED** [review]

File: test/api.test.ts (hunk 2 above)
Comment (hossam-nasr) [reply test/api.test.ts]: Done

**Review by nturinski — APPROVED** [review]

**Pull request merged**